### DISCUSSION TITLE
Failed to load fonts.

### DISCUSSION THREAD

**Discussion by RetrogisusDEV** [CloneWith/opsu#125] [Bug report]
**Version:** 1.3.0
**Build date:** 2024-07-27 09:03
**OS:** Windows 10 (amd64)
**JRE:** 22.0.2
**OpenGL version:** 1.1.0 (Microsoft Corporation)
**Error:** `Failed to load fonts.`
**Stack trace:**
```
java.lang.NullPointerException: Cannot invoke "java.util.Map.toString()" because "clonewith.opsu.I18N.tlMap" is null
	at clonewith.opsu.I18N.getFontmap(I18N.java:95)
	at org.newdawn.slick.UnicodeFont.addLocaleGlyphs(UnicodeFont.java:378)
	at itdelatrisu.opsu.ui.Fonts.loadFont(Fonts.java:104)
	at itdelatrisu.opsu.ui.Fonts.init(Fonts.java:83)
	at itdelatrisu.opsu.Utils.init(Utils.java:136)
	at itdelatrisu.opsu.states.Splash.init(Splash.java:109)
	at org.newdawn.slick.state.StateBasedGame.init(StateBasedGame.java:171)
	at org.newdawn.slick.AppGameContainer.setup(AppGameContainer.java:393)
	at itdelatrisu.opsu.Container.runSetup(Container.java:113)
	at itdelatrisu.opsu.Container.start(Container.java:72)
	at itdelatrisu.opsu.Opsu.main(Opsu.java:190)
```

I get this error when trying to run OPSU and I don't know what is missing or what is needed, I installed the fonts inside the jar, (although I had a problem with the FFMPEG and Graphics libraries that were not extracted, which I have already solved)

My question: what is the problem?

(Translator of Google xd)

**Comment by CloneWith** [comment]
It seems that opsu! cannot read the translation strings correctly. Even I've encountered this a few times...

Currently the I18N implementation is a bit messed up, and I'm trying to fix it. You can use older versions or compile from source code for now.

**Reply by RetrogisusDEV** [reply]
Understood, I'll try it in a bit. The device's language is set to ES_421 (Latin American Spanish), and it seems to only have neutral (global) languages available. It's strange that even when loading the default (English), it gives that error...

**Comment by CloneWith** [comment]
Now it should be fixed in v1.3.1, have a try!

**Reply by RetrogisusDEV** [reply]
The previous error is gone, but now I'm getting a "Failed to load 'Menu-Background'" error. I'm trying to figure out the cause. Should the skin images be placed in the 'natives' folder or in a separate folder? I'll look through the code now to see if I can solve it through brute force.

**Reply by CloneWith** [reply]
Are you using the default skin? I found some problems exposed in the log files while running the game. Will look into these.
